Source/Urho3D/UI/LineEdit.cpp

```diff
@@ -599,10 +599,11 @@ void LineEdit::UpdateText()
     }
     else
     {
-        String echoText{};
-
-        for (unsigned i{ 0 }; i < utf8Length; ++i)
-            echoText.AppendUTF8(echoCharacter_);
+        const String echoText{ [&]{
+                String text{};
+                for (unsigned i{ 0 }; i < utf8Length; ++i)
+                    text.AppendUTF8(echoCharacter_);
+                return text; }() };

```

Comment:
Why? Is this a trolling attempt?

Reply:
Are you overlooking the `const String echoText{ [&]{` part?

Reply:
What does that achieve besides hurting code readability and debug build performance?

Reply:
From the CppCoreGuidelines:
> 28: Use lambdas for complex initialization, especially of const variables

Some place else:
> Const is a powerful tool for writing safer code, and it can help compiler optimizations. You should use it as much as possible...